### ISSUE TITLE
Guzzle conflict with Brizy Pro plugin

### ISSUE THREAD

**Issue by dougaitken** [woocommerce/google-listings-and-ads#845]
### Describe the bug:

When Google Listings & Ads is active, after activating [Brizy Pro](https://www.brizy.io/brizy-pro-pricing/) an error occurs on all GLA admin pages, and a log file is created.

Deactivating Brizy Pro resolves the issue.

28819703-hc

### Expected behavior:

As the Guzzle error looks to be resolved in `1.1.0` via #751, then no Guzzle errors should happen

### Actual behavior:

Fatal error: 

<details>

```
2021-06-30T07:31:24+00:00 CRITICAL Uncaught Error: Call to undefined function Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\stream_for() in /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:400
Stack trace:
#0 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php(59): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\CurlFactory->applyHandlerOptions(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\EasyHandle), Array)
#1 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/CurlHandler.php(43): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\CurlFactory->create(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#2 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php(28): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\CurlHandler->__invoke(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#3 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/Proxy.php(48): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\Proxy::Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\{closure}(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#4 /srv/htdocs/wp-content/plugins/google-listings-and-ads/src/Internal/DependencyManagement/GoogleServiceProvider.php(215): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\Proxy::Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Handler\{closure}(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#5 /srv/htdocs/wp-content/plugins/google-listings-and-ads/src/Internal/DependencyManagement/GoogleServiceProvider.php(184): Automattic\WooCommerce\GoogleListingsAndAds\Internal\DependencyManagement\GoogleServiceProvider->Automattic\WooCommerce\GoogleListingsAndAds\Internal\DependencyManagement\{closure}(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#6 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/PrepareBodyMiddleware.php(35): Automattic\WooCommerce\GoogleListingsAndAds\Internal\DependencyManagement\GoogleServiceProvider->Automattic\WooCommerce\GoogleListingsAndAds\Internal\DependencyManagement\{closure}(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#7 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Middleware.php(31): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\PrepareBodyMiddleware->__invoke(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#8 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/RedirectMiddleware.php(71): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Middleware::Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\{closure}(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#9 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/HandlerStack.php(75): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\RedirectMiddleware->__invoke(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#10 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Client.php(331): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\HandlerStack->__invoke(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#11 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Client.php(168): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Client->transfer(Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Request), Array)
#12 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Client.php(187): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Client->requestAsync('GET', Object(Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Psr7\Uri), Array)
#13 /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/ClientTrait.php(44): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Client->request('GET', 'https://api-vip...', Array)
#14 /srv/htdocs/wp-content/plugins/google-listings-and-ads/src/API/Google/Connection.php(111): Automattic\WooCommerce\GoogleListingsAndAds\Vendor\GuzzleHttp\Client->get('https://api-vip...')
#15 /srv/htdocs/wp-content/plugins/google-listings-and-ads/src/API/Site/Controllers/Google/AccountController.php(113): Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Connection->get_status()
#16 /wordpress/core/5.7.2/wp-includes/rest-api/class-wp-rest-server.php(1139): Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Google\AccountController->Automattic\WooCommerce\GoogleListingsAndAds\API\Site\Controllers\Google\{closure}(Object(WP_REST_Request))
#17 /wordpress/core/5.7.2/wp-includes/rest-api/class-wp-rest-server.php(985): WP_REST_Server->respond_to_request(Object(WP_REST_Request), '/wc/gla/google/...', Array, NULL)
#18 /wordpress/core/5.7.2/wp-includes/rest-api/class-wp-rest-server.php(412): WP_REST_Server->dispatch(Object(WP_REST_Request))
#19 /wordpress/core/5.7.2/wp-includes/rest-api.php(354): WP_REST_Server->serve_request('/wc/gla/google/...')
#20 /wordpress/core/5.7.2/wp-includes/class-wp-hook.php(292): rest_api_loaded(Object(WP))
#21 /wordpress/core/5.7.2/wp-includes/class-wp-hook.php(316): WP_Hook->apply_filters(NULL, Array)
#22 /wordpress/core/5.7.2/wp-includes/plugin.php(551): WP_Hook->do_action(Array)
#23 /wordpress/core/5.7.2/wp-includes/class-wp.php(388): do_action_ref_array('parse_request', Array)
#24 /wordpress/core/5.7.2/wp-includes/class-wp.php(750): WP->parse_request('')
#25 /wordpress/core/5.7.2/wp-includes/functions.php(1291): WP->main('')
#26 /wordpress/core/5.7.2/wp-blog-header.php(16): wp()
#27 /wordpress/core/5.7.2/index.php(17): require('/wordpress/core...')
#28 {main}
  thrown in /srv/htdocs/wp-content/plugins/google-listings-and-ads/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php on line 400
```

</details>

### Steps to reproduce:
1. Install Google Listings & Ads - no need to setup yet
2. Install Brizy from WordPress.org and then install Brizy Pro (ping me internally for a zip)
3. Go through the GLA onboarding and notice a toast error message indicating an error
4. Deactivate Brizy Pro and try again
5. Head to **WooCommerce > Status > Logs > Select - Latest Fatal Error Log under drop-down (right side) > Click - View button** to see log file

**Comment by mikkamp** [comment]
Possible related issue reported here: https://wordpress.org/support/topic/fatal-error-3906/
Although we can't confirm if version 1.1 is being used there. Will need to dig deeper, but in #751 we chose to only prefix the common libraries we use, so there might be additional ones which conflict.

**Comment by mikkamp** [comment]
I was able to track this down because the deprecated functions are loaded conditionally in `vendor/guzzlehttp/psr7/src/functions_include.php`:

```
// Don't redefine the functions if included multiple times.
if (!function_exists('GuzzleHttp\Psr7\str')) {
    require __DIR__ . '/functions.php';
}
```

When another plugin (brizy-pro in this case) defines `GuzzleHttp\Psr7\str` the functions.php file is no longer loaded and the error occurs. To fix this we should also ensure those calls are prefixed. Doing so is a bit more challenging because the composer.json loads the `functions_include.php` and those are each in a subdirectory, `guzzlehttp/composer.json` doesn't exist.

However this issue is no longer present with the latest trunk, because of #830, there we no longer need the deprecated functions, the ones in the `Util` class are used instead.

**Comment by macflyhub** [comment]
I had "GTG Product Feed for Shopping" Version 1.2.2 | By GutenGeek 
installed. Same bug. Disabled and working now.

**Comment by tijosh** [comment]
Original reporter notes a lingering fatal error using Brizy Pro plugin while on most current version of GLA (1.3.0):

> PHP message: PHP Fatal error: Uncaught Error: Call to undefined function Automattic\\WooCommerce\\GoogleListingsAndAds\\Vendor\\GuzzleHttp\\Psr7\\uri_for() in /var/www/html/wp-content/plugins/google-listings-and-ads/vendor/google/gax/src/RequestBuilder.php:233

Can you take another look at that @mikkamp ?

4179454-zen

**Comment by mikkamp** [comment]
Thanks for the report I was able to reproduce the issue, it turns out there were still some deprecated functions being used by another library, which is resolved in #915
